providers/cloudflare/rest.go

```diff
@@ -88,6 +88,32 @@ func (c *CloudflareApi) deleteRec(rec *cfRecord, domainID string) *models.Correc
 	}
 }
 
+func (c *CloudflareApi) createZone(domainName string) (string, error) {
+	type createZone struct {
+		Name string `json:"name"`
+	}
+	endpoint := zonesURL
```

Comment:
probably don't need to redeclare endpoint as a variable. Just use zonesURL the one place you use endpoint later.